### DISCUSSION TITLE
Image processing failure shouldn't break the build

### DISCUSSION THREAD

**Discussion by antoinerousseau** [gatsbyjs/gatsby#30707] [Help]
## Description

When building for production, the build fails with:

```
error Failed to process image /drone/src/.cache/caches/default-site-plugin/9087f2e81eeec4fa7fe75cbb49505ca9/4eaff2f4-4d62-4166-a779-c144f7a67cd8.png.
It is probably corrupt, so please try replacing it. If it still fails, please open an issue with the image attached.

Error: pngload: non-recoverable state
vips2png: unable to write to target target
```

(also same error on my macbook)

### Steps to reproduce

Try loading this image with the build-in gatsby sharp plugin (through `createRemoteFileNode` and a `childImageSharp` query):

![4eaff2f4-4d62-4166-a779-c144f7a67cd8](https://user-images.githubusercontent.com/160279/112859064-c5150680-90b2-11eb-84b8-6514ae15f198.png)

Note that there is a `try/catch` around my `createRemoteFileNode` but it doesn't help.

On a side note, running `pngcheck` on that file outputs `invalid bKGD length`.

### Expected result

Load/transform the image, or ignore it with a warning.

### Actual result

Break the build.

### Environment

```
  System:
    OS: macOS 11.2.3
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.15.1 - ~/.nvm/versions/node/v14.15.1/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 6.14.8 - ~/.nvm/versions/node/v14.15.1/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 89.0.4389.90
    Firefox: 84.0.2
    Safari: 14.0.3
  npmPackages:
    gatsby: ^2.30.3 => 2.32.11
    gatsby-image: ^2.9.0 => 2.11.0
    gatsby-plugin-google-gtag: ^2.6.0 => 2.8.0
    gatsby-plugin-manifest: ^2.3.4 => 2.12.1
    gatsby-plugin-react-helmet: ^3.2.3 => 3.10.0
    gatsby-plugin-react-svg: ^3.0.0 => 3.0.0
    gatsby-plugin-sharp: ^2.5.5 => 2.14.3
    gatsby-plugin-sitemap: ^2.4.12 => 2.12.0
    gatsby-plugin-styled-components: ^3.2.2 => 3.10.0
    gatsby-plugin-typescript: ^2.3.2 => 2.12.1
    gatsby-source-filesystem: ^2.2.3 => 2.11.1
    gatsby-source-graphql: ^2.7.3 => 2.14.0
    gatsby-transformer-sharp: ^2.4.5 => 2.12.1
```

**Comment by LekoArts** [comment]
Hi Antoine!

Thanks for doing the `pngcheck` also locally -  so `sharp` is working as expected there then. You can use the `failOnError` option if you want to only emit a warning: https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-sharp#options

**Reply by antoinerousseau** [reply]
Hi @LekoArts, I'm sorry but I forgot to mention that `failOnError` is already set to `false` in my sharp plugin config!

```
    {
      resolve: `gatsby-plugin-sharp`,
      options: {
        defaultQuality: 75,
        failOnError: false,
      },
    },
```

But it still fails

**Reply by seekidsdream** [reply]
Were you ever able to solve this? I also have failOnError = false in my config, but still getting failed builds. Any suggestions?
